src/coreclr/tools/aot/crossgen2/crossgen2_inbuild.csproj

```
@@ -12,9 +12,14 @@
     -->
     <PublishAot Condition="$(NETCoreSdkPortableRuntimeIdentifier.StartsWith('win-'))">true</PublishAot>
     <PublishSingleFile Condition="'$(PublishAot)' != 'true'">true</PublishSingleFile>
+    <TrimmerSingleWarn>false</TrimmerSingleWarn>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(CrossBuild)' == 'true' and '$(StageOneBuild)' != 'true' and '$(StageTwoBuild)' != 'true'">
```

```suggestion
  <PropertyGroup Condition="'$(StageOneBuild)' != 'true' and '$(StageTwoBuild)' != 'true'">
```
We can still end up building this for the vertical in what runtime would not consider a cross-build (technically we build it in source-build too).

---

Multistage is not designed for source-build or tested with SB at all. Lets fix the issue at hand and worry about enabling multistage for source-build (which highly doubt is necessary) separately.

---

The problem is the existing condition won't be satisfied when CrossBuild is not true, and it needs to be.